code/__DEFINES/directions.dm

```diff
@@ -28,3 +28,6 @@
 #define IS_DIR_DIAGONAL(dir) (dir & (dir - 1))
 /// returns TRUE if direction is cardinal and false if not
 #define IS_DIR_CARDINAL(dir) (!IS_DIR_DIAGONAL(dir))
+
+/// Inverse direction, taking into account UP|DOWN if necessary.
+#define REVERSE_DIR(dir) ( ((dir & 85) << 1) | ((dir & 170) >> 1) )
```

Comment:
this is horribly cursed, I love it

Reply:
(it's incredible, isn't it?) The REVERSE_DIR changes are from #25550 and will hopefully disappear here once that gets merged and I update this branch.